Framework/Algorithms/src/ConjoinWorkspaces.cpp

```diff
@@ -71,25 +73,24 @@ void ConjoinWorkspaces::exec() {
     AnalysisDataService::Instance().remove(getPropertyValue("InputWorkspace2"));
     // Set the result workspace to the first input
     setProperty("InputWorkspace1", output);
-    return;
-  }
-
-  // Check that the input workspaces meet the requirements for this algorithm
-  this->validateInputs(*ws1, *ws2);
-
-  if (this->getProperty("CheckOverlapping")) {
-    this->checkForOverlap(*ws1, *ws2, true);
-    m_overlapChecked = true;
+  } else {
+	  // Check that the input workspaces meet the requirements for this algorithm
+	  this->validateInputs(*ws1, *ws2);
+
+	  if (this->getProperty("CheckOverlapping")) {
+		  this->checkForOverlap(*ws1, *ws2, true);
+		  m_overlapChecked = true;
+	  }
+
+	  MatrixWorkspace_sptr output = execWS2D(*ws1, *ws2);
+	  // Copy the history from the original workspace
+	  output->history().addHistory(ws1->getHistory());
+
+	  // Delete the second input workspace from the ADS
+	  AnalysisDataService::Instance().remove(getPropertyValue("InputWorkspace2"));
+	  // Set the result workspace to the first input
+	  setProperty("InputWorkspace1", output);
   }
-
-  MatrixWorkspace_sptr output = execWS2D(*ws1, *ws2);
-  // Copy the history from the original workspace
-  output->history().addHistory(ws1->getHistory());
-
-  // Delete the second input workspace from the ADS
-  AnalysisDataService::Instance().remove(getPropertyValue("InputWorkspace2"));
-  // Set the result workspace to the first input
-  setProperty("InputWorkspace1", output);
 }
```

Comment:
Have you done anything besides replace (if { code, return } more_code) with if {code} else {more_code} ?